embassy-stm32/src/spi/mod.rs

```diff
@@ -180,6 +180,18 @@ impl<'d, T: Instance, Tx, Rx> Spi<'d, T, Tx, Rx> {
         )
     }
 
+    /// Useful for on chip peripherals like SUBGHZ which are hardwired.
+    /// The bus can optionally be exposed externally with `Spi::new()` still.
+    pub fn new_internal(
```

Comment:
Should be marked as pub(crate) IMO since we likely want the HAL to have some interface for those special peripherals?